examples/posix_sockets/udp.c

```diff
@@ -98,6 +118,19 @@ static int udp_send(char *addr_str, char *port_str, char *data, unsigned int num
     src.sin6_family = AF_INET6;
     dst.sin6_family = AF_INET6;
     memset(&src.sin6_addr, 0, sizeof(src.sin6_addr));
+    /* parse interface id */
+#ifdef SOCK_HAS_IPV6
+    char *iface;
+    uint32_t netif;
+    iface = ipv6_addr_split_iface(addr_str); /* also removes interface id */
+    if (!iface) {
+        netif = _get_first_netif();
+    }
+    else {
+        netif = atoi(iface);
+    }
```

Comment:
I think you want to replicate what [`ping6`]( https://github.com/RIOT-OS/RIOT/blob/master/sys/shell/commands/sc_gnrc_icmpv6_echo.c#L196) does. (Would probably be a good idea to move that to common code).

Otherwise you might always set the `sin6_scope_id` for a global address to the first interface.

Reply:
Yes, I found that code snipplet in RIOT, in [examples/gnrc_networking](https://github.com/RIOT-OS/RIOT/blob/01d0e30229f1f80d622abd28bac178115066e0f8/examples/gnrc_networking/udp.c#L47). 

After some thought I am also not happy with that solution. It would be better to avoid RIOT internal functions as much as possible here. Maybe I remove the `_get_first_netif()` function , since that is a convenience feature. 
The user can/should specify an interface explicitly when using link-local addresses, which is the standard for most Unix commands.